zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala

```diff
@@ -134,7 +134,7 @@ class SQLPPGenerator extends AsterixQueryGenerator {
 
   private def parseLookup(lookups: Seq[LookupStatement],
                           exprMap: Map[String, FieldExpr]): ParsedResult = {
-    val producedExprs = mutable.Map.newBuilder[String, FieldExpr]
+    val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
```

Comment:
LinkedHashMap will preserve the order of which the fields are added.
Thus, we can ensure fields in producedExprs have the same order as the fields in group or select statements.